docker/Dockerfile

```diff
@@ -14,19 +14,48 @@ RUN apt-get update \
     # Install google-fluentd at version 1.5.35-1.
     && curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | REPO_SUFFIX=20180810-1 DO_NOT_INSTALL_CATCH_ALL_CONFIG=true bash \
     # Clean up to reduce image size.
-    && apt-get remove -y \
+    && apt-get purge -y \
         curl \
         lsb-release \
         build-essential \
+        gtk2.0 \
+        adduser \
+        libkrb5-3 \
     && rm -rf /var/lib/apt/lists/* /tmp/* \
-    && apt-get autoremove -y \
+    && apt-get autoremove -y --purge \
     && find /var/log -name "*.log" -type f -delete \
+    # Remove .a .c .cc .h files and include libraries
+    && rm -rf /opt/google-fluentd/embedded/include /opt/google-fluentd/embedded/lib/*.a \
+    && (find /opt/google-fluentd/embedded/ -name '*.c' -o -name '*.cc' -o -name '*.h'  | xargs rm) \
     # Remove docs.
-    && rm -rf /usr/share/doc /usr/share/man /opt/google-fluentd/embedded/share/doc /opt/google-fluentd/embedded/lib/ruby/gems/*/doc ./opt/google-fluentd/embedded/lib/ruby/*/rdoc /opt/google-fluentd/embedded/share/gtk-doc /opt/google-fluentd/embedded/share/man /opt/google-fluentd/embedded/share/terminfo \
-    # Remove gems that are only needed when ingesting logs to Treasure Data Cloud.
-    && /opt/google-fluentd/embedded/bin/gem uninstall td td-client td-logger fluent-plugin-td fluent-plugin-td-monitoring hirb parallel ohai mixlib-cli mixlib-config mixlib-log mixlib-shellout systemu -ax --force \
+    && rm -rf \
+        /usr/share/doc \
+        /usr/share/man \
+        /opt/google-fluentd/embedded/lib/ruby/gems/*/doc \
+        /opt/google-fluentd/embedded/share/doc \
+        /opt/google-fluentd/embedded/share/gtk-doc \
+        /opt/google-fluentd/embedded/share/man \
+        /opt/google-fluentd/embedded/share/terminfo \
+    # Remove unused gems.
+    && /opt/google-fluentd/embedded/bin/gem uninstall -ax --force \
+        # Gems for ingesting logs to Treasure Data Cloud.
+        td \
+        td-client \
+        td-logger \
+        fluent-plugin-td \
+        fluent-plugin-td-monitoring \
+        hirb \
+        parallel \
+        ohai \
+        mixlib-cli \
+        mixlib-config \
+        mixlib-log \
+        mixlib-shellout \
+        systemu \
     # Remove unused api client libraries.
-    && cd /opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/gems/google-api-client-0.23.4/generated/google/apis && ls -1 /opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/gems/google-api-client-0.23.4/generated/google/apis | grep -v 'logging' | xargs rm -rf \
+    && cd /opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/gems/google-api-client-0.23.4/generated/google/apis \
```

Comment:
The subshell should start here. However, this might be better expressed as:
```
    && (find /opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/gems/google-api-client-0.23.4/generated/google/apis -depth 1 \! -name logging | xargs rm -rf)
```

Reply:
Changed.